### PR TITLE
Add new facility details sidebar

### PR DESCRIPTION
**Pull request by TaiWilkin**
## Overview

In order to allow maintenance work to be released while the Extended
Profile feature is in development, the new sidebar is wrapped in a
feature flag. When the flag is disabled, the old sidebar is displayed.

This initial implementation of the ExtendedProfile sidebar has only a
limited subset of fields and behaviors implemented. Specifically,
the claim status header; the OAR ID; the facility name, address, and
location; the static map; the contributors; and the action buttons have
been implemented.

Extended fields, claim fields, contributor/nonstandard fields, the
closed facility status and activity information, and the 'additional
information' popups have not yet been implemented.

Connects #1539 

## Demo
Unclaimed
Screen Shot 2021-12-08 at 3 23 05 PM
Pending
Screen Shot 2021-12-08 at 3 22 49 PM
Embedded
Screen Shot 2021-12-08 at 3 21 57 PM
Claimed
Screen Shot 2021-12-08 at 4 03 36 PM
Screen Shot 2021-12-08 at 4 04 28 PM
Screen Shot 2021-12-08 at 4 04 43 PM

## Notes

We have followup cards for adding the 'additional information' popups and extended fields, and for adding the closed facility status and activity information. 

We should create an additional task to add contributor / nonstandard fields. 

The embedded map view of the sidebar wasn't presented in the wireframes. In the original embedded map view, we replace the action buttons with a link to the OAR; remove the contributors list; and hide 'other addresses', 'other locations', and so on. I've replicated this in the new sidebar. 

The header was a particular special case now that it shows claim information; based on previous requests from clients regarding the embedded map mode, clients prefer no information regarding other contributors to be displayed in their maps, and we also want no buttons to things like claims in embed mode. I've instead displayed 'contributed by (embedded map contributor)' in the header. 

There is a special state in the original sidebar where a contributor has claimed a facility but that claim is pending. During this state, other users can still access the 'claim this facility' button, but the pending claimant sees a 'pending claim' message. I've also replicated this in the new sidebar. 

## Testing Instructions

* Run `./scripts/update`
* View the details for a facility and confirm the sidebar behaves as before. 
* Turn on the `extended_profile` switch in the Django admin. Also give contributor 2 embedded map permissions. 
* Login as `c2@example.com` and view an unclaimed facility with multiple addresses/etc in the sidebar. Confirm that it matches [the wireframes](https://app.abstract.com/projects/441633f0-7d8c-11e9-af1e-972cbddca60a/branches/df31450b-d39a-4306-803d-c0d926ffd046/commits/72f5e69bf7a06d99182d72a68d99a26dfba4e1e8/files/fc1791e3-b8d2-4182-b240-fef32fabe978/layers/0EE5B838-BBA3-4D30-B62E-6127BF3E87B7?collectionId=1390af9f-d01b-4801-95a8-d8d14937a521&collectionLayerId=ffae49a3-4f5b-450a-bb3a-1926431af19e&mode=build) and that the data presented is correct. Confirm that the back button, contributor profile buttons, 'report a facility', and 'edit' buttons work as expected.
* View a facility with no 'other addresses' etc data and confirm the facility details do not display the additional information arrows. 
* View the multi-address facility in your embedded map preview. Adjust the color and font variables to confirm they work as expected. Confirm that the main header button and the list items are not clickable. 
* In regular view, claim one facility using the header button and one facility using the 'claim this facility' action button at the bottom of the sidebar. 
* As an admin user, approve one of the two claims.
* As `c2@example.com`, view the pending facility and confirm that the header shows a pending message and the 'claim this facility' action button is not displayed. 
* View the approved claim facility and confirm that the header shows the claim information and is not clickable (other than the back button). The 'claim this facility' and 'dispute claim' buttons shouldn't appear.
* As `c3@example.com`, view the pending facility - the claim buttons should be present. 
* As `c3@example.com`, view the approved claim - the 'dispute claim' button should be present and the header should show the claim information. 
* Mark a facility as having inexact_coordinates in the Django admin, then view the inexact coordinates message in full view and in embedded mode in the sidebar. 
* View the sidebar in mobile size and ensure it displays properly. 

## Checklist

- [x] `fixup!` commits have been squashed
- [x] CI passes after rebase
- [x] CHANGELOG.md updated with summary of features or fixes, following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines

### CODE REVIEW

**Review by jwalgran — COMMENTED** [review]
This is an excellent start and thank you so much for the comprehensive testing instructions.

I only found one data issue related to inactive or private contributions.

**Comment by lederer** [comment]
Added a small commit with some polish:

> - Lowered opacity of divider lines in profile sidebar.
> - Some type, icon, and spacing tweaks.
> - Changed a `<ul>` to a `<div>` bc its children aren't `<li>`s.

**Comment by lederer** [comment]
I've created a FontAwesome **kit** using our Pro subscription. 

Put in `<head>`:

```html
<script src="https://kit.fontawesome.com/ae021710c1.js" crossorigin="anonymous"></script>
```

You can then use this to render the icons:
```html
<i class="far fa-fw fa-store-slash"></i>
```

That will use the correct weight (`far`) and make them fixed width so they all line up nicely.

"Dispute claim" should use `shield-alt` while "Claim" should use `shield-check`.

**Comment by lederer** [comment]
I'm having trouble completing the testing instructions…

Eg, when I click "Claim this facility" (while logged in as `c2@`) I get a full-page error:

```
Uncaught TypeError: Cannot read properties of undefined (reading 'style')
    at setTransform (DomUtil.js:216)
    at NewClass.<anonymous> (Map.js:1622)
    at NewClass.fire (Events.js:190)
    at NewClass._moveEnd (Map.js:1259)
    at NewClass._resetView (Map.js:1194)
    at NewClass.setView (Map.js:206)
    at NewClass.setZoom (Map.js:218)
    at NewClass._updateZoomLevels (Layer.js:273)
    at NewClass._removeZoomLimit (Layer.js:245)
    at NewClass.onRemove (GridLayer.js:173)
    at NewClass.removeLayer (Layer.js:185)
    at NewClass.removeFrom (Layer.js:66)
    at NewClass.remove (Layer.js:59)
    at NewClass.remove (Map.js:798)
    at Map.componentWillUnmount (Map.js:270)
    at callComponentWillUnmountWithTimer (react-dom.development.js:19580)
    at HTMLUnknownElement.callCallback (react-dom.development.js:188)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:237)
    at invokeGuardedCallback (react-dom.development.js:292)
    at safelyCallComponentWillUnmount (react-dom.development.js:19587)
    at commitUnmount (react-dom.development.js:20109)
    at commitNestedUnmounts (react-dom.development.js:20163)
    at unmountHostComponents (react-dom.development.js:20443)
    at commitDeletion (react-dom.development.js:20500)
    at commitMutationEffects (react-dom.development.js:22782)
    at HTMLUnknownElement.callCallback (react-dom.development.js:188)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:237)
    at invokeGuardedCallback (react-dom.development.js:292)
    at commitRootImpl (react-dom.development.js:22509)
    at unstable_runWithPriority (scheduler.development.js:653)
    at runWithPriority$1 (react-dom.development.js:11039)
    at commitRoot (react-dom.development.js:22381)
    at finishSyncRender (react-dom.development.js:21807)
    at performSyncWorkOnRoot (react-dom.development.js:21793)
    at react-dom.development.js:11089
    at unstable_runWithPriority (scheduler.development.js:653)
    at runWithPriority$1 (react-dom.development.js:11039)
    at flushSyncCallbackQueueImpl (react-dom.development.js:11084)
    at flushSyncCallbackQueue (react-dom.development.js:11072)
    at discreteUpdates$1 (react-dom.development.js:21893)
    at discreteUpdates (react-dom.development.js:806)
    at dispatchDiscreteEvent (react-dom.development.js:4168)
```

**Comment by lederer** [comment]
I just noticed that my polish commit broke the build. Will fix soon.

**Comment by jwalgran** [comment]
I don't think it was your commit, @lederer. I think there may be a problem on Jenkinds that is affeecting multiple projects. I was able to checkout this PR and ruccessfully run `./scripts/cibuild`

**Comment by lederer** [comment]
Oh weird. I assumed it was that commit bc the build was ✅ on previous commits. Thanks for investigating.

**Comment by TaiWilkin** [comment]
Rebased to handle merge conflict and add in the changes on develop.

File: src/app/src/components/FontAwesomeIcon.jsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { loadCSS } from 'fg-loadcss/src/loadCSS';
+import Icon from '@material-ui/core/Icon';
+
+class FontAwesome extends React.Component {
```
[the remark this answers is not in the document]
Comment (TaiWilkin) [reply src/app/src/components/FontAwesomeIcon.jsx]: Removed this file as unnecessary following the switch to the pro FontAwesome kit from Scott.

**Comment by TaiWilkin** [comment]
> Eg, when I click "Claim this facility" (while logged in as `c2@`) I get a full-page error:
> 
> ```
> Uncaught TypeError: Cannot read properties of undefined (reading 'style')
>     at setTransform (DomUtil.js:216)
> ```

This is an existing issue on develop. It appears when we click away from the map in some instances, but only in debug (so not only staging or production) - you can click the 'x' on the error to get away from the error message and the app behaves as expected. 

I thought we had created an issue for this, but looking for it again I didn't find one. I've added a new [bug issue](https://github.com/open-apparel-registry/open-apparel-registry/issues/1563).

File: src/django/api/serializers.py [hunk 2]
```diff
@@ -800,6 +808,15 @@ def sort_order(k):
...
         return grouped_data
 
+    def get_created_from(self, facility):
+        user_can_see_detail = can_user_see_detail(self)
+        list_item = facility.created_from
+        return {
+            'created_at': list_item.created_at,
+            'contributor': get_contributor_name(list_item.source.contributor,
+                                                user_can_see_detail)
+        }
```
[the remark this answers is not in the document]
Comment (TaiWilkin) [reply src/django/api/serializers.py]: I've updated to check if there are any `FacilityMatches` with `should_display_association` for the `facility.created_from` prior to displaying the name. I also filtered out contributors from the Contributors section whose details shouldn't be visible, and then hid the section if there were no visible contributors.

**Review by jwalgran — APPROVED** [review]
The updates look good. I retested private and inactive sources and an inactive match and verified that in all three cases the contributor name was anonymized.

**Comment by TaiWilkin** [comment]
Thank you for reviewing!

**Pull request merged**